source/Pages/Notification/components/ImportToken/index.jsx

```diff
@@ -0,0 +1,127 @@
+import { reviewPendingTransaction } from '@modules/storageManager';
+import DisplayBox from '../Sign/components/Details/components/DisplayBox';
+
+i18n.use(initReactI18next).init(initConfig);
+
+const portRPC = new PortRPC({
+  name: 'notification-port',
+  target: 'bg-script',
+  timeout: 20000,
+});
+
+portRPC.start();
+
+const ImportToken = ({ args, metadata, callId, portId, setOnTimeout, transactionId, }) => {
+  const classes = useStyles();
+  const { t } = useTranslation();
+  const dispatch = useDispatch();
+
+  const [loading, setLoading] = useState('');
+  const [handled, setHandled] = useState(false);
+
+  const { url, icons } = metadata || {};
+  const { canisterId, symbol, standard, logo } = args || {};
+
+  const handleImportToken = async (status) => {
+    setLoading(status);
+    reviewPendingTransaction(transactionId, () => { });
+    const success = await portRPC.call('handleRequestImportToken', [
+      { status, token: { canisterId, standard, logo } },
+      callId,
+      portId,
+      transactionId,
+    ]);
+    setHandled(true);
+    if (success) {
+      window.close();
+    }
+    setLoading('');
+  };
+
+  useEffect(() => {
+    setOnTimeout(() => () => {
+      handleImportToken(CONNECTION_STATUS.refused);
+    });
+    sendMessage({ type: HANDLER_TYPES.GET_STATE, params: {} }, (state) => {
+      if (state?.wallets?.length) {
+        dispatch(setAccountInfo(state.wallets[state.currentWalletId]));
+      }
+    });
+  }, []);
+
+  window.onbeforeunload = () => {
+    if (!handled) {
+      handleImportToken(CONNECTION_STATUS.refused);
+    }
+  };
+
+  return (
+    <Layout disableProfile disableNavigation incStatus>
+      <div className={classes.padTop}>
+        <Container className={classes.container}>
+          <IncomingAction url={url} image={icons[0] || null} action={t('addToken.importTitle')} />
+
+          <DisplayBox
+            title={symbol}
+            subtitle={canisterId}
+            img={<TokenIcon
+              image={logo}
+              className={classes.tokenImage}
+              symbol={symbol}
+            />}
+          />
+
+          <div className={classes.buttonContainer}>
+            <Button
+              variant="default"
+              value={t('common.decline')}
+              onClick={() => handleImportToken(CONNECTION_STATUS.refused)}
+              style={{ width: '96%' }}
+              loading={loading === CONNECTION_STATUS.refused}
+              disabled={loading === CONNECTION_STATUS.accepted}
+              fullWidth
+            />
+            <Button
+              variant="rainbow"
+              value={t('common.allow')}
+              onClick={() => handleImportToken(CONNECTION_STATUS.accepted)}
+              fullWidth
+              style={{ width: '96%' }}
+              loading={loading === CONNECTION_STATUS.accepted}
+              disabled={loading === CONNECTION_STATUS.refused}
+              wrapperStyle={{ textAlign: 'right' }}
```

Comment:
you could abstract most of these props to like `commonButtonProps` or something like that. 

Also 
`const isRefused = loading === CONNECTION_STATUS.refused;`
`const isAccepted = loading === CONNECTION_STATUS.accepted;`